kerneltest/vmtest.sh

```diff
@@ -78,23 +115,63 @@ check_executable() {
     fi
 }
 
+choose_kernel() {
+    # Run the tests.
+    kernel_versions=("5.4" "5.10" "5.19" "6.1")
+    arch_versions=("arm64")
+
+    for kernel in "${kernel_versions[@]}"; do
+        for arch in "${arch_versions[@]}"; do
+            use_kernel "$kernel" "$arch"
+            # Ensure that the adaptive unwind shard mechanism
+            # works in memory constrained environments.
+            if [[ "$kernel" == "5.4" ]]; then
+                #vm_run_x86 "$kernel" "0.7G" "$arch"
+                vm_run_arm "$kernel" "0.7G" "$arch"
+            else
+                #vm_run_x86 "$kernel" "1.5G" "$arch"
+                vm_run_arm "$kernel" "1.5G" "$arch"
+            fi
+        done
+    done
+
+}
+
 run_tests() {
     # Initial checks.
     check_executable "curl"
     check_executable "qemu-system-x86_64"
+    check_executable "qemu-system-aarch64"
 
     # Run the tests.
     kernel_versions=("5.4" "5.10" "5.19" "6.1")
-
-    for kernel in "${kernel_versions[@]}"; do
-        use_kernel "$kernel"
-        # Ensure that the adaptive unwind shard mechanism
-        # works in memory constrained environments.
-        if [[ "$kernel" == "5.4" ]]; then
-            vm_run "$kernel" "0.7G"
-        else
-            vm_run "$kernel" "1.5G"
-        fi
+    arch_versions=("amd64", "arm64")
+
+    for arch in "${arch_versions[@]}"; do
+        for kernel in "${kernel_versions[@]}"; do
+            use_kernel "$kernel" "$arch"
+            # Ensure that the adaptive unwind shard mechanism
+            # works in memory constrained environments.
+            if [[ "$kernel" == "5.4" ]]; then
```

Comment:
As discussed on our 1:1, let's reduce the if/else complexity here and use the arch passed to `vm_run`